proxy/src/transport/tls/config.rs

```diff
@@ -143,8 +147,30 @@ impl CommonConfig {
         let private_key = load_file_contents(&settings.private_key)?;
         let private_key = untrusted::Input::from(&private_key);
 
-        // `CertResolver::new` is responsible for the consistency check.
-        let cert_resolver = CertResolver::new(&root_cert_store, cert_chain, private_key)?;
+        // Ensure the certificate is valid for the services we terminate for
+        // TLS. This assumes that server cert validation does the same or
+        // more validation than client cert validation.
+        //
+        // XXX: Rustls currently only provides access to a
+        // `ServerCertVerifier` through
+        // `rustls::ClientConfig::get_verifier()`.
+        //
+        // XXX: Once `rustls::ServerCertVerified` is exposed in Rustls's
+        // safe API, remove the `map(|_| ())` below.
+        //
+        // TODO: Restrict accepted signatutre algorithms.
+        let certificate_was_validated =
+            rustls::ClientConfig::new().get_verifier().verify_server_cert(
+                    &root_cert_store,
+                    &cert_chain,
+                    (settings.service_identity.0).0.as_ref(),
```

Comment:
NIT: there's a lot going on here; seems like it might be clearer if this was a variable that was passed to `verify_server_cert`?

Reply:
It would be nice to name the `.0` refs here, perhaps by exposing `fn`s on each type with an appropriate name?

Reply:
:+1:, I nearly commented on that as well.

Reply:
I factored out a variable `let identity_dns_name = (settings.service_identity.0).0.as_ref();` I would have liked to write `let Identity(Arc(DnsName(identity_dns_name))) = (settings.service_identity.0).0.as_ref();` but we can't pattern match through `Arc`. We could definitely add a named accessor for the `.0` fields but AFAICT this is the only place that would ever need it so I think we're better off without it.